### ISSUE TITLE
Self-update not working on SunOS since nzbhydra2 5.x

### ISSUE THREAD

**Issue by hubert3** [theotherp/nzbhydra2#857]
The built-in update function used to work fine in nzbhydra2 4.x but has not since I upgraded to 5.x

Running 5.1.2 recently (generic release type launched from py3 launcher), the update function seems to download nzbhydra2-5.1.6-amd64-linux.zip which is not correct

Extract from wrapper.log:

```2023-03-01 21:57:58,505  INFO - Determined release type: generic
2023-03-01 21:57:58,587  INFO - Determined java version as '17' from version string 'openjdk version "17.0.5" 2022-10-18'
2023-03-01 21:57:58,588  INFO - Starting NZBHydra main process with command line: java -Xmx256M -DfromWrapper=true -DinternalApiKey=anjjdrigcwzojgmzipyd -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=xxx/nzbhydra2/data/logs -Dspring.output.ansi.enabled=ALWAYS -Dsun.security.pkcs11.enable-solaris=false -jar xxx/nzbhydra2/lib/core-5.1.2-exec.jar --datafolder xxx/nzbhydra2/data in folder xxx/nzbhydra2
2023-03-01 21:58:30,941  INFO - Determined process URI to be http://127.0.0.1:5076/
2023-03-01 21:59:14,172  INFO - NZBHydra main process has terminated for updating
2023-03-01 21:59:14,174  WARNING - lib folder and core(.exe) found. Either delete the executable to use the generic release type (using java and ignoring the executable) or delete the lib folder to use the ex
ecutable and not require java
2023-03-01 21:59:14,174  INFO - Determined release type: generic
2023-03-01 21:59:14,175  INFO - Extracting updated files to xxx/nzbhydra2
2023-03-01 21:59:21,245  INFO - Removing update ZIP xxx/nzbhydra2/data/update/nzbhydra2-5.1.6-amd64-linux.zip
2023-03-01 21:59:21,252  INFO - Updating lib folder for generic release type
2023-03-01 21:59:21,253  INFO - Found 1 JAR files in lib folder
2023-03-01 21:59:21,253  INFO - Found file: core-5.1.2-exec.jar
2023-03-01 21:59:21,254  WARNING - New JAR file in lib folder is the same as the old one. The update may not have found a newer version or failed for some reason
2023-03-01 21:59:21,254  INFO - Deleting folder xxx/nzbhydra2/data/update
2023-03-01 21:59:21,260  INFO - Update successful, restarting Hydra main process
```

PS this issue on SunOS has also not been fixed yet https://github.com/theotherp/nzbhydra2/issues/831 

I am running with a modified wrapper with the following 2 changes (disabling the wrapper check to skip the warning):

```
--- nzbhydra2wrapperPy3.py      Thu. Mar.  2 21:41:18 2023
+++ nzbhydra2wrapperPy3.py.SunOS        Mon. Feb.  6 11:06:59 2023
@@ -25,6 +24,7 @@
 from logging.handlers import RotatingFileHandler
 from enum import Enum

+os.environ["NZBHYDRA_DISABLE_WRAPPER_CHECK"] = "1"

 class ReleaseType(str, Enum):
     NATIVE = "native"
@@ -445,6 +435,8 @@
     if args.debug:
         java_arguments.append("-Ddebug=true")

+    java_arguments.append("-Dsun.security.pkcs11.enable-solaris=false")
+
     if releaseType == ReleaseType.NATIVE:
         arguments = [args.java] + java_arguments + arguments
     else:
```

**Comment by theotherp** [comment]
The log says `lib folder and core(.exe) found. Either delete the executable to use the generic release type (using java and ignoring the executable) or delete the lib folder to use the executable and not require java`

So you should delete the `core` file and see if that helps.

**Comment by hubert3** [comment]
@theotherp I have tried that, even with only lib/core-5.1.x-exec.jar being present it is still downloading and unzipping nzbhydra2-5.1.6-amd64-linux.zip which is not correct for my platform

The wrapper correctly determines that the running release type is generic, so shouldn't it be downloading nzbhydra2-5.1.6-generic.zip?

**Comment by theotherp** [comment]
You're right, that's not correct.

**Comment by cocokola** [comment]
I followed the instructions to download the latest build, killed the hung process, unzipped overwriting files,   rebooted, and have the exact same result.  The utility launches but nothing is running.  the only way forward without diving into install logs or monitoring task is trying to launch the app again.   Which I did again to get to this thread in the updates list.  what is needed to fix?
i cant be the only user with this issue
